framework/Models/Code.py

```diff
@@ -308,8 +312,11 @@ def createNewInput(self,currentInput,samplerType,**kwargs):
     if not os.path.exists(subDirectory):
       os.mkdir(subDirectory)
     for index in range(len(newInputSet)):
-      newInputSet[index].setPath(subDirectory)
-      shutil.copy(self.oriInputFiles[index].getAbsFile(),subDirectory)
+      subSubDirectory = os.path.join(subDirectory,newInputSet[index].subDirectory)
+      if newInputSet[index].subDirectory.strip() != "" and not os.path.exists(subSubDirectory):
+        os.mkdir(subSubDirectory)
```

Comment:
Is it important to create the directory in two different places? I am just wondering if between `createNewInput` and `initialize`, one of these will always get to this line first? This would essentially make the other one into dead code. I don't know which is the "right" place and if there are use cases where both may occur.